docker-compose.build.yml

```diff
@@ -0,0 +1,71 @@
+version: '3'
+services:
+  # Server
+  server_npm:
+    build:
+      context: 'server/'
+      dockerfile: 'Dockerfile.packages'
+    image: liberoadmin/reviewer_server_packages:${IMAGE_TAG}
+    networks: 
+        - 'reviewer_build'
+  server_typescript:
+    build:
+      context: 'server/'
+      dockerfile: "Dockerfile.typescript"
+      args:
+        image_tag: ${IMAGE_TAG}
+    image: liberoadmin/reviewer_server_typescript:${IMAGE_TAG}
+    depends_on:
+      - server_npm
+    networks: 
+        - 'reviewer_build'
+  reviewer_server:
+    build:
+      context: 'server/'
+      dockerfile: "Dockerfile.application"
+      args:
+        image_tag: ${IMAGE_TAG}
+    image: liberoadmin/reviewer_server:${IMAGE_TAG}
+    depends_on:
+      - server_npm
+      - server_typescript
+    networks: 
+        - 'reviewer_build'
+
+  # Client
+  client_npm:
+    build:
+      context: 'client/'
+      dockerfile: 'Dockerfile.packages'
+    image: liberoadmin/reviewer_client_packages:${IMAGE_TAG}
+    networks: 
+        - 'reviewer_build'
+
+  client_webpack:
+    build:
+      context: 'client/'
+      dockerfile: 'Dockerfile.webpack'
+      args:
+        image_tag: ${IMAGE_TAG}
+    image: liberoadmin/reviewer_client_webpack:${IMAGE_TAG}
+    depends_on:
+      - client_npm
+    networks: 
+        - 'reviewer_build'
+
+  reviewer_client:
+    build:
+      context: 'client/'
+      dockerfile: "Dockerfile.application"
+      args:
+        image_tag: "${IMAGE_TAG}"
+    image: liberoadmin/reviewer_client:${IMAGE_TAG}
+    depends_on:
+      - client_npm
+      - client_webpack
+    networks: 
+        - 'reviewer_build'
+networks:
```

Comment:
what's the purpose of a custom external network as opposed to the default one created and managed by `docker-compose`?

Reply:
For some reason we had an issue on Mac's ... worked fine in Linux :man_shrugging:

Reply:
@petereast .. although it is giving this error in CI:

```
ERROR: Network reviewer_build declared as external, but could not be found. Please create the network manually using `docker network create reviewer_build` and try again.
make[1]: *** [lint_server] Error 1
```

Reply:
if it's `external` it has to be created manually, so Travis CI fails when it hasn't got it; suggest to find the root cause as `docker-compose` would normally manage all networks and create/delete them as needed. Usually the problem is outdated state from a old `docker-compose.yml` file that has created resources that are not in the configuration anymore.

Reply:
The problem occurred with Docker for Mac, where the parallel execution resulted in `docker-compose` creating the network two times simultaneously causing an error because when it tried to use it, it couldn't decide which one to use

Reply:
I think I've seen this before, `docker-compose` not being robust to parallel execution. But it shouldn't be creating networks during `build` only during `up` right?
Use two different `docker-compose.yml` files if these are two independent services?